System-wide sqlite 3.36 breaks installation of R package (which comes with sqlite 3.46)

OS - Rocky Linux 8 x86_64, R 4.4.0 (built from source) using standard gcc 8.5.0

Installing 2.3.6 from CRAN is fine.

Installing 2.3.7 from CRAN fails with this error:

```
In file included from ext-series.c:5:
vendor/extensions/series.c: In function ‘seriesBestIndex’:
vendor/extensions/series.c:500:13: error: ‘SQLITE_INDEX_CONSTRAINT_LIMIT’ undeclared (first use in this function); did you mean ‘SQLITE_INDEX_CONSTRAINT_LIKE’?
     if( op>=SQLITE_INDEX_CONSTRAINT_LIMIT
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
             SQLITE_INDEX_CONSTRAINT_LIKE
vendor/extensions/series.c:500:13: note: each undeclared identifier is reported only once for each function it appears in
vendor/extensions/series.c:501:13: error: ‘SQLITE_INDEX_CONSTRAINT_OFFSET’ undeclared (first use in this function); did you mean ‘SQLITE_INDEX_CONSTRAINT_LIKE’?
      && op<=SQLITE_INDEX_CONSTRAINT_OFFSET
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
             SQLITE_INDEX_CONSTRAINT_LIKE
vendor/extensions/series.c: At top level:
vendor/extensions/series.c:617:3: warning: excess elements in struct initializer
   0                          /* xIntegrity */
   ^
vendor/extensions/series.c:617:3: note: (near initialization for ‘seriesModule’)
make: *** [/opt/bioit/R-core/4.4.0/lib64/R/etc/Makeconf:195: ext-series.o] Error 1
ERROR: compilation failed for package ‘RSQLite’
```

--- a/.Rbuildignore
+++ b/.Rbuildignore
@@ -41,3 +41,5 @@ a.out.dSYM
 ^\.gitpod\.yml$
 ^src/CMakeLists\.txt$
 ^patch$
+^compile_commands\.json$
+^\.cache$

--- a/.gitignore
+++ b/.gitignore
@@ -22,3 +22,5 @@ inst/doc
 CRAN-RELEASE
 tests/testthat/testthat-problems.rds
 CRAN-SUBMISSION
+/compile_commands.json
+/.cache/

--- a/src/Makevars
+++ b/src/Makevars
@@ -1,5 +1,6 @@
 # BOOST_NO_AUTO_PTR: https://github.com/eddelbuettel/bh/issues/57#issuecomment-406261595
-PKG_CPPFLAGS=-DRSQLITE_USE_BUNDLED_SQLITE \
+PKG_CPPFLAGS=-Ivendor \
+             -DRSQLITE_USE_BUNDLED_SQLITE \
              -DSQLITE_ENABLE_RTREE \
              -DSQLITE_ENABLE_FTS3 \
              -DSQLITE_ENABLE_FTS3_PARENTHESIS \

--- a/src/Makevars
+++ b/src/Makevars
@@ -1,6 +1,5 @@
 # BOOST_NO_AUTO_PTR: https://github.com/eddelbuettel/bh/issues/57#issuecomment-406261595
-PKG_CPPFLAGS=-I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 \
-             -DRSQLITE_USE_BUNDLED_SQLITE \
+PKG_CPPFLAGS=-DRSQLITE_USE_BUNDLED_SQLITE \
              -DSQLITE_ENABLE_RTREE \
              -DSQLITE_ENABLE_FTS3 \
              -DSQLITE_ENABLE_FTS3_PARENTHESIS \

--- a/src/RSQLite_types.h
+++ b/src/RSQLite_types.h
@@ -1,7 +1,7 @@
 #ifndef __RSQLITE_TYPES__
 #define __RSQLITE_TYPES__
 
-#include <RSQLite.h>
+#include "RSQLite.h"
 
 #include "DbConnection.h"
 #include "DbResult.h"

